### PR TITLE
Location types perms

### PR DESCRIPTION
**Pull request by emord**
@esoergel  - Here's the first part of the locations work. Can't access locations types if you're assigned to a location

@snopoke

### CODE REVIEW

**Comment by emord** [comment]
Note merging into other dev branch

**Comment by benrudolph** [comment]
:+1:

File: corehq/apps/locations/permissions.py [hunk 1]
```diff
@@ -42,3 +42,22 @@ def _inner(request, domain, loc_id, *args, **kwargs):
                 return view_fn(request, domain, loc_id, *args, **kwargs)
         raise Http404()
     return locations_access_required(_inner)
+
+
+def user_can_edit_location_types(user, project):
+    if not project.location_restriction_for_users:
+        return True
+
+    return False if user.get_domain_membership(project.name).location_id else True
```
Comment (snopoke) [inline corehq/apps/locations/permissions.py]: Could just be `return not user.get_domain_membership(project.name).location_id`

**Comment by snopoke** [comment]
Looks good to me but will leave @esoergel to merge.

**Pull request merged**